src/Kernel32.Shared/Win32Exception.cs

```diff
@@ -46,7 +46,7 @@ public Win32Exception()
         /// <param name="error">The Win32 error code associated with this exception.</param>
         public Win32Exception(Win32ErrorCode error)
 #if DESKTOP
-            : base((int)error)
+            : this(error, error.GetMessage() ?? $"Unknown error (0x{(int)error:x8})")
```

Comment:
I've merged all the changes except this one, since it was huge and I wanted to avoid merge conflicts, and this last change we can debate independently and merge if it's deemed good.